primitives/Cargo.toml

```
@@ -34,121 +23,60 @@ crypto_kx = { version = "0.2.1", features = ["serde"] }
+jf-vdf = { path = "../vdf", default-features = false }
+jf-vid = { path = "../vid", default-features = false }
 rand_chacha = { workspace = true }
 rayon = { version = "1.5.0", optional = true }
 serde = { workspace = true }
 sha2 = { workspace = true }
-sha3 = { workspace = true }
-tagged-base64 = { workspace = true }
 zeroize = { version = "1.5", default-features = false }
 
 [dev-dependencies]
-ark-bls12-377 = { workspace = true }
-ark-bls12-381 = { workspace = true }
-ark-bn254 = { workspace = true }
-ark-bw6-761 = { workspace = true }
+ark-ed-on-bls12-377 = "0.4.0"
+ark-ed-on-bls12-381 = "0.4.0"
 ark-ed-on-bls12-381-bandersnatch = "0.4.0"
+ark-ed-on-bn254 = "0.4.0"
 bincode = "1.3"
-criterion = "0.5.1"
-hashbrown = "0.14.3"
-sha2 = { version = "0.10.1" }
-
-[[bench]]
-name = "merkle_path"
-harness = false
-
-[[bench]]
-name = "pcs"
-harness = false
-required-features = ["test-srs"]
-
-[[bench]]
-name = "pcs-size"
-path = "benches/pcs_size.rs"
-harness = false
-required-features = ["test-srs"]
-
-[[bench]]
-name = "kzg-gpu"
-path = "benches/kzg_gpu.rs"
-harness = false
-required-features = ["test-srs"]
-
-[[bench]]
-name = "reed-solomon"
-path = "benches/reed_solomon.rs"
-harness = false
-
-[[bench]]
-name = "advz"
-harness = false
-required-features = ["test-srs"]
-
-[[bench]]
-name = "minroot"
-harness = false
-
-[[bench]]
-name = "bls-signature"
-path = "benches/bls_signature.rs"
-harness = false
 
 [features]
 default = ["parallel"]
 std = [
-        "ark-bls12-377/std",
-        "ark-bls12-381/std",
-        "ark-bn254/std",
-        "ark-bw6-761/std",
         "ark-std/std",
         "ark-serialize/std",
         "ark-ff/std",
         "ark-ec/std",
-        "ark-poly/std",
         "ark-crypto-primitives/std",
-        "merlin/std",
-        "num-bigint/std",
-        "num-traits/std",
         "sha2/std",
-        "sha3/std",
         "zeroize/std",
-        "itertools/use_std",
         "rand_chacha/std",
         "jf-utils/std",
         "jf-relation/std",
+        "jf-rescue/std",
+        "jf-merkle-tree/std",
+        "jf-vdf/std",
 ]
-print-trace = ["ark-std/print-trace"]
-kzg-print-trace = [
-        "print-trace",
-] # leave disabled to reduce pollution in downstream users of KZG (such as VID)
+gadgets = [
+        "jf-relation",
+        "jf-rescue/gadgets",
+        "jf-signature/gadgets",
+        "jf-merkle-tree/gadgets",
+]
+print-trace = ["ark-std/print-trace", "jf-vid/print-trace"]
 parallel = [
         "ark-ff/parallel",
         "ark-ec/parallel",
         "jf-utils/parallel",
         "jf-relation/parallel",
+        "jf-vid/parallel",
+        "jf-pcs/parallel",
         "rayon",
 ]
-test-srs = []
-seq-fk-23 = [] # FK23 without parallelism
-icicle = [
-       "icicle-cuda-runtime",
-       "icicle-core",
-       "icicle-core/arkworks",
-       "icicle-bn254",
-       "parallel",
-]
-gpu-vid = ["icicle"]
+test-srs = ["jf-pcs/test-srs", "jf-vid/test-srs"]
+gpu-vid = ["jf-pcs/icicle", "jf-vid/gpu-vid"]
```

```suggestion
gpu-vid = ["jf-vid/gpu-vid"]
```
the former is already implied by the latter

---

removed [ac0c37b](https://github.com/EspressoSystems/jellyfish/pull/556/commits/ac0c37bcb8daaef34ea55be9e6456d78f8359216)